reviewfile.cs

```diff
@@ -0,0 +1,387 @@
+        private readonly ILogger<OperationController> mylogger;
+        private readonly IUserContext _userContext;
+        private readonly IValidator<InitiationRequestAndContext> _validator;
+        private readonly IMediator _mediator;
+        public readonly IAsyncRepository<SGPrepOperation> _repo;      
+        private readonly IBusinessDateDistanceCalculator _distanceCalculator;
+        private readonly IRelativeEastCoastDateTimeConverter _relativeEastCoastDateTimeConverter;
+        private readonly IERMDistributionListService _ERMDistributionListService;
+        private readonly IOperationWorkingDaysService _operationWorkingDaysService;
+        private readonly IOperationInfoService _operationInfoRetriever;
+        private readonly IDocumentService _documentService;
+        private readonly IMilestonesService _milestonesService;
+        private readonly ISGPrepUserRoleService _sgPrepUserRoleService;
+
+        public OperationController(ISGPrepInScopeAssessorService inScopeAssessor,
+            ILogger<OperationController> logger,
+            IUserContext userContext,
+            IMediator mediatr,
+            IValidator<InitiationRequestAndContext> validator,
+            IAsyncRepository<SGPrepOperation> repo,
+            IClock clock,
+            IBusinessDateDistanceCalculator distanceCalculator,
+            IRelativeEastCoastDateTimeConverter relativeEastCoastDateTimeConverter,
+            IERMDistributionListService ERMDistributionListService,
+            IOperationWorkingDaysService operationWorkingDaysService,
+            IOperationInfoService prepInfoRetriever,
+            IDocumentService documentService,
+            IMilestonesService milestonesService,
+            ISGPrepUserRoleService sgPrepUserRoleService
+            )
+        {
+            _inScopeAssessor = inScopeAssessor;
+            mylogger = logger;
+            _userContext = userContext;
+            _mediator = mediatr;
+            _validator = validator;
+            _repo = repo;
+            _clock = clock;
+            _distanceCalculator = distanceCalculator;
+            _relativeEastCoastDateTimeConverter = relativeEastCoastDateTimeConverter;
+            _ERMDistributionListService = ERMDistributionListService;
+            _operationWorkingDaysService = operationWorkingDaysService;
+            _operationInfoRetriever = prepInfoRetriever ?? throw new ArgumentNullException(nameof(prepInfoRetriever));
+            _documentService = documentService;
+            _milestonesService = milestonesService;
+            _sgPrepUserRoleService = sgPrepUserRoleService;
+        }
+
+        [HttpPost("{opNumber}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> Get(string opNumber)
```

Comment:
This parameter "opNumber" is used everywhere without validating if it's null or empty, might want to add that validation everywhere